### ISSUE TITLE
Passing /nocopyright from bash command line returns an error.

### ISSUE THREAD

**Issue by marcelodeaguiar** [dotnet/codeformatter#56]
Passing */nocopyright* from a bash command line (like git bash) returns the error: **Error: The specified rule type was not find:** _path-to-file/nocopyright_

I think it has something to do with the ```StringComparer.OrdinalIgnoreCase``` used to parse the arguments.

**Comment by jaredpar** [comment]
I just used this switch today on some code I was working on and it functioned as expected.  Could you show me the exact command line you were using?

**Comment by marcelodeaguiar** [comment]
Sure,

![error](https://cloud.githubusercontent.com/assets/1771472/6210753/da505d22-b5b9-11e4-9cdc-07af36ca40b1.png)

I made some tests and it has nothing to do with the comparer, but with the way git bash interprets my commands. the argument is passed as **C:/Program Files (x86)/Git/nocopyright** intead of **/nocopyright** problably because of the **/**.

Maybe you should change params to start with dash **-**.

**Comment by khellang** [comment]
Isn't this a shell problem rather than a bug in the formatter?

**Comment by marcelodeaguiar** [comment]
@khellang I don't know if its _bug_. Isn't it the way unix based shells work? Git bash is a fairly common shell so I think it should be considered...

**Comment by khellang** [comment]
Well, you might argue that `/` isn't the best way to denote a switch/flag cross-plat. I guess that's why most UNIX stuff use `-` or `--` :smile:

I'm not sure there's a way around it other than switching the character, since Git Bash ends up passing a path. Maybe do the check using `.EndsWith`? :stuck_out_tongue_closed_eyes:

**Comment by marcelodeaguiar** [comment]
@khellang 
I think the best approach is to use `--` as its pretty common and its an easy solution. 
`.EndsWith` looks more like a bad workaround I think.

I can submit the change if it is accepted.

**Comment by khellang** [comment]
Maybe something like [Mono.Options](https://www.nuget.org/packages/Mono.Options/) or similar could be used?

**Comment by jaredpar** [comment]
I think we need to move to accept `-` as an option specifier.

**Comment by richlander** [comment]
@jaredpar - I suspect that's right. After using git all the time, the dashes are starting to feel quite natural.

**Comment by marcelodeaguiar** [comment]
Most of the tools use 2 dashes for option full name and 1 dash for option shortcut. You can see this all over Git documentation.

@khellang I did not know [Mono.Options](https://www.nuget.org/packages/Mono.Options/) but it looks awesome.

**Comment by khellang** [comment]
@marcelodeaguiar Yep, it's a quick way to get decent command line options (supporting `-`, `--` and `/` switches) with minimal fuzz (I think its just a single .cs file). :smile:

**Comment by GaTechThomas** [comment]
It seems that more detail in the error message would be helpful.

**Comment by jaredpar** [comment]
@GaTechThomas what extra data could help here?

**Comment by GaTechThomas** [comment]
I don't use unix, so I can't reproduce, but from the screenshot it looks like the error output is a bit confusing. Maybe showing something along the lines of "Executing Codeformatter.exe using the following parameters: ...".

That output may not be desirable for all runs, but maybe so when the -verbose option is set.

**Comment by jaredpar** [comment]
@GaTechThomas but what makes CodeFormatter special here?  Why shouldn't all console apps have this output?

**Comment by GaTechThomas** [comment]
I would expect that any app writing verbose output would be better served by displaying this sort of information. 

It's the equivalent of, say, asking a contractor to do a series of things around the house, and then having them repeat back in detail what they're going to do... Sometimes things get lost in how they are asked or how they're heard, or both, and both parties are better off for having clarified the scope of work.
